### ISSUE TITLE
dotnet 3 sdk snap *** stack smashing detected ***

### ISSUE THREAD

**Issue by paaland** [dotnet/runtime#3792]
# dotnet 3 sdk snap *** stack smashing detected ***

 **OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64**

Install dotnet-sdk via snap:

```
$ snap install dotnet-sdk --classic
$ snap alias dotnet-sdk.dotnet dotnet
```

Create new console application
```
$ dotnet new console
$ dotnet run
```

Should write "Hello World", but instead ends up with

*** stack smashing detected ***: <unknown> terminated

# General

```
$ which dotnet
/snap/bin/dotnet
```

```
$ snap --version

Debian 10 buster
snap    2.41
snapd   2.41
series  16
debian  10
kernel  4.19.0-6-amd64
```

```
$ dotnet -info

.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64
 Base Path:   /snap/dotnet-sdk/49/sdk/3.0.100/

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  3.0.100 [/snap/dotnet-sdk/49/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0 [/snap/dotnet-sdk/49/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0 [/snap/dotnet-sdk/49/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by martin-cermak** [comment]
We are also facing *** stack smashing detected ***: <unknown> terminated

We are not using SNAP. Application runs in Docker. Currently we cant caugh when it happends, no debug, no exception.

**Comment by misterspeedy** [comment]
Hit identical issue on Ubuntu 20.4.2:

Repro, brand new Ubuntu 20.4.2 VM:

    sudo snap install dotnet-sdk --classic --channel=3.1
    sudo snap alias dotnet-sdk.dotnet dotnet
    mkdir demo
    cd demo
    dotnet new console -lang F#
    dotnet run
    *** stack smashing detected ***: terminated

**Comment by rubenpetrosyan1** [comment]
> 
> 
> We are also facing *** stack smashing detected ***: terminated
> 
> We are not using SNAP. Application runs in Docker. Currently we cant caugh when it happends, no debug, no exception.
> 
> Environment: .NET Core 3.1, Debian 10.3, Docker version 19.03.8

hi martin-cermak
did you resolve the issue?
we are running into the same issue with dotnet build inside docker container.

**Comment by martin-cermak** [comment]
Hi Ruben,
we spent a lot of time with diagnostics and I have only ideas where bug is related. At first I must say that our app had bugs. BUT on Windows, app was crashed and shown exception.

On Linux runtime there was this strange "stack smashing". It was related to more things, I thing a combination of async call / event rutine where was some function called on NULL object. On windows there was right exception about NULL object, but on Linux it crashed.

So in our solution, after many hours we changed this:
SomeObject.CallFunction()

to

if (SomeObject != null) {
  SomeObject.CallFunction()      // or simplier version using ?
}


I thing it was also not related to docker but linux runtime itself...

Martin
